dpnp/backend/extensions/window/kaiser.cpp

```diff
@@ -0,0 +1,180 @@
+        using dpnp::kernels::i0::impl::cyl_bessel_i0;
+#endif
+
+        const auto i = id.get(0);
+        const T alpha = (N - 1) / T(2);
+        const T tmp = (i - alpha) / alpha;
+        data[i] = cyl_bessel_i0(beta * sycl::sqrt(1 - tmp * tmp)) /
+                  cyl_bessel_i0(beta);
+    }
+};
+
+template <typename T, template <typename> class Functor>
+sycl::event kaiser_impl(sycl::queue &q,
+                        char *result,
+                        const std::size_t nelems,
+                        const float beta,
+                        const std::vector<sycl::event> &depends)
+{
+    dpctl::tensor::type_utils::validate_type_for_device<T>(q);
+
+    T *res = reinterpret_cast<T *>(result);
+
+    sycl::event kaiser_ev = q.submit([&](sycl::handler &cgh) {
+        cgh.depends_on(depends);
+
+        using KaiserKernel = Functor<T>;
+        cgh.parallel_for<KaiserKernel>(sycl::range<1>(nelems),
+                                       KaiserKernel(res, nelems, beta));
+    });
+
+    return kaiser_ev;
+}
+
+template <typename fnT, typename T>
+struct KaiserFactory
+{
+    fnT get()
+    {
+        if constexpr (std::is_floating_point_v<T>) {
+            return kaiser_impl<T, KaiserFunctor>;
+        }
+        else {
+            return nullptr;
+        }
+    }
+};
+
+std::pair<sycl::event, sycl::event>
+    py_kaiser(sycl::queue &exec_q,
+              const float beta,
+              const dpctl::tensor::usm_ndarray &result,
+              const std::vector<sycl::event> &depends)
+{
+    dpctl::tensor::validation::CheckWritable::throw_if_not_writable(result);
+
+    int nd = result.get_ndim();
+    if (nd != 1) {
+        throw py::value_error("Array should be 1d");
+    }
+
+    if (!dpctl::utils::queues_are_compatible(exec_q, {result.get_queue()})) {
+        throw py::value_error(
+            "Execution queue is not compatible with allocation queue.");
+    }
+
+    const bool is_result_c_contig = result.is_c_contiguous();
+    if (!is_result_c_contig) {
+        throw py::value_error("The result input array is not c-contiguous.");
+    }
+
+    size_t nelems = result.get_size();
+    if (nelems == 0) {
+        return std::make_pair(sycl::event{}, sycl::event{});
+    }
+
+    int result_typenum = result.get_typenum();
+    auto array_types = dpctl_td_ns::usm_ndarray_types();
+    int result_type_id = array_types.typenum_to_lookup_id(result_typenum);
+    auto fn = kaiser_dispatch_vector[result_type_id];
+
+    if (fn == nullptr) {
+        throw std::runtime_error("Type of given array is not supported");
+    }
+
+    char *result_typeless_ptr = result.get_data();
+    sycl::event kaiser_ev =
+        fn(exec_q, result_typeless_ptr, nelems, beta, depends);
+    sycl::event args_ev =
+        dpctl::utils::keep_args_alive(exec_q, {result}, {kaiser_ev});
+
+    return std::make_pair(args_ev, kaiser_ev);
+}
+
+void init_kaiser_dispatch_vectors()
+{
+    dpctl_td_ns::DispatchVectorBuilder<kaiser_fn_ptr_t, KaiserFactory,
```

Comment:
[Here](https://github.com/IntelPython/dpnp/compare/impl-kaiser...reuse-init-window-dispatch?expand=1) is an example how we can reuse from "common.hpp" in case it might be helpful.